u8timeseries/backtesting/backtesting.py

```diff
@@ -259,3 +260,193 @@ def plot_residuals_analysis(residuals: TimeSeries, num_bins: int = 20):
+    val_series
+        A TimeSeries instance used for validation during model tuning and for training during evaluation.
+    test_series
+        A TimeSeries instance used for validation when evaluating a model.
+    width
+        An integer indicating the number of plots that are displayed in one row.
+    metric:
+        A function that takes two TimeSeries instances as inputs and returns a float error value.
+    verbose:
+        Whether to print progress.
+    """
+
+    raise_if_not(width > 1, "Please choose an integer 'width' value larger than 1", logger)
+
+    # list of tuples containing model classes and hyperparameter selection, if required
+    model_parameter_tuples = [
+        (NaiveSeasonal, {
+            'K': list(range(1, 31))
+        }),
+        (FFT, {
+            'nr_freqs_to_keep': [5, 10, 25, 50, 100, 150, 200],
+            'trend': [None, 'poly', 'exp']
+        }),
+        (Theta, {
+            'theta': [0, 1] + list(range(3, 50))
+        }),
+        (Prophet, {}),
+        (AutoARIMA, {})
+    ]
+
+    # set up plot grid
+    height = math.ceil(len(model_parameter_tuples) / width) + 1
+    fig = plt.figure(constrained_layout=True, figsize=(width * 4, height * 4))
+    gs = fig.add_gridspec(height, width)
+
+    # arrays to store data for bar charts
+    mape_vals = []
+    times = []
+    index = []
+
+    # iterate through model type selection
+    iterator = _build_tqdm_iterator(model_parameter_tuples, verbose)
+    for i, (model_class, params) in enumerate(iterator):
+        
+        # if necessary, tune hyperparameters using train_series and val_series
+        if (len(params.keys()) > 0):
+            model = backtest_gridsearch(model_class, params, train_series, val_series, metric=metric)
+        else:
+            model = model_class()
+
+        # fit on train and val series, predict test series
+        train_and_val_series = train_series.append(val_series)
+        start_time = time.time()
+        model.fit(train_and_val_series)
+        end_time = time.time()
+        predictions = model.predict(len(test_series))
+
+        # plot predictions against observations
+        y = int(i / width)
+        x = i - y * width
+        ax = fig.add_subplot(gs[y, x])
+        mape_val = metrics.mape(predictions, test_series)
+        ax.set_title(str(model) + "\n" + metric.__name__ + ": " + str(round(mape_val, 2)))
+        train_and_val_series[-len(test_series):].plot(label='train and val', ax=ax)
+        test_series.plot(label='test', ax=ax)
+        predictions.plot(label='pred', ax=ax)
+        plt.legend()
+
+        # record data for bar charts
+        mape_vals.append(mape_val)
+        times.append(end_time - start_time)
+        index.append(str(model)[:3] + "..")
+
+    # plot bar charts
+    ax_mapes = fig.add_subplot(gs[-1, 0])
+    ax_mapes.set_title("Performance")
+    ax_mapes.set_ylabel(metric.__name__ + ' value')
+    pd.Series(mape_vals, index=index).plot.bar(ax=ax_mapes, rot=0)
```

Comment:
does this way of plotting print the actual numeric values? I think it'd be nice for users to be able to read the actual numbers (not just see them) when comparing metrics and execution time.

Reply:
There does not seem to be a built-in function like that for bar plots. Currently I am printing the values of the metric and the runtime in the plot title of the respective model. Do you think that's ok?